scripts/release-tasks.ts

```diff
@@ -61,113 +53,66 @@ export async function runReleaseTasks(opts: BuildOptions, args: ReadonlyArray<st
-              // https://github.com/sindresorhus/np/pull/73#discussion_r72385685
-              if (err.stdout !== '' || err.stderr !== '') {
-                throw err;
-              }
-            },
-          ),
-      skip: () => isDryRun,
-    },
-    {
-      title: 'Check current branch',
-      task: () =>
-        execa('git', ['symbolic-ref', '--short', 'HEAD']).then(({ stdout }) => {
-          if (stdout !== 'main' && !isAnyBranch) {
-            throw new Error('Not on `main` branch. Use --any-branch to publish anyway.');
-          }
-        }),
-      skip: () => isDryRun || opts.isCI, // in CI, we may be publishing from another branch
-    },
-    {
-      title: 'Check local working tree',
-      task: () =>
-        execa('git', ['status', '--porcelain']).then(({ stdout }) => {
-          if (stdout !== '') {
-            throw new Error('Unclean working tree. Commit or stash changes first.');
-          }
-        }),
-      skip: () => opts.isCI, // skip in CI, as we may have files staged needed to publish
-    },
-    {
-      title: 'Check remote history',
-      task: () =>
-        execa('git', ['rev-list', '--count', '--left-only', '@{u}...HEAD']).then(({ stdout }) => {
-          if (stdout !== '0' && !isAnyBranch) {
-            throw new Error('Remote history differs. Please pull changes.');
-          }
-        }),
-      skip: () => isDryRun || opts.isCI, // no need to check remote history in CI, we just pulled it
-    },
-  );
+  tasks.push({
+    /**
+     * When we both pre-release and release, it's beneficial to ensure that the tag does not already exist in git.
+     * Doing so ought to catch out of the ordinary circumstances that ought to be investigated.
+     */
+    title: 'Check git tag existence',
+    task: () =>
+      execa('git', ['fetch'])
+        // Retrieve the prefix for a version string - https://docs.npmjs.com/cli/v7/using-npm/config#tag-version-prefix
+        .then(() => execa('npm', ['config', 'get', 'tag-version-prefix']))
+        .then(
+          ({ stdout }) => (tagPrefix = stdout),
+          () => {},
+        )
+        // verify that a tag for the new version string does not already exist by checking the output of
+        // `git rev-parse --verify`
+        .then(() => execa('git', ['rev-parse', '--quiet', '--verify', `refs/tags/${tagPrefix}${newVersion}`]))
+        .then(
+          ({ stdout }) => {
+            if (stdout) {
+              throw new Error(`Git tag \`${tagPrefix}${newVersion}\` already exists.`);
+            }
+          },
+          (err) => {
+            // Command fails with code 1 and no output if the tag does not exist, even though `--quiet` is provided
+            // https://github.com/sindresorhus/np/pull/73#discussion_r72385685
+            if (err.stdout !== '' || err.stderr !== '') {
+              throw err;
+            }
+          },
+        ),
+    skip: () => isDryRun,
```

Comment:
Non blocking suggestion to make it also a bit more readable using `async/await` instead of promise chaining:

```suggestion
    task: async () => {
      await execa('git', ['fetch'])
      // Retrieve the prefix for a version string - https://docs.npmjs.com/cli/v7/using-npm/config#tag-version-prefix
      const { stdout: tagPrefix } = await execa('npm', ['config', 'get', 'tag-version-prefix'])
        .catch(() => ({}))
      // verify that a tag for the new version string does not already exist by checking the output of
      // `git rev-parse --verify`
      const { stdout: hasGitTag } = await execa('git', ['rev-parse', '--quiet', '--verify', `refs/tags/${tagPrefix}${newVersion}`]).catch((err) => {
        // Command fails with code 1 and no output if the tag does not exist, even though `--quiet` is provided
        // https://github.com/sindresorhus/np/pull/73#discussion_r72385685
        if (err.stdout !== '' || err.stderr !== '') {
          throw err;
        }
      });
      if (hasGitTag) {
        throw new Error(`Git tag \`${tagPrefix}${newVersion}\` already exists.`);
      }
    },
    skip: () => isDryRun,
```

Reply:
I'm gonna hold off on this for the moment - I want to see if we can eliminate `Listr` here entirely. I started to do this (`async` work) and a few `if` statement reorders and felt like I was optimizing something that might be a better off being rewritten with some testing under it (and I didn't get that far on JEDI days 😉)